liferay-headless/liferay-portal
# LPS-199304 | Add new GET EndPoint that retrieve a single entry by ID to the UI

**Description:** This PR adds the capability of creating a GET endpoint that retrieves a single element given a path parameter configurable by the UI.
Also, it extracts logic of Filters and Sorts to a component.

**How to test it:**
1. Add API Builder Feature Flags
```
#API BUILDER FEATURE FLAGS
feature.flag.LPS-186757=true
feature.flag.LPS-178642=true
feature.flag.LPS-184413=true
feature.flag.LPS-167253=true
```
2. Create an object
     2.1. Go to Application Menu
     2.2. Create new object called Student (Plural name: students)
     2.3. Go to the newly created object > Fields
     2.4. Create a new field called "carnet" of type text and check "Accept Unique Values Only"
     2.5. Create other field called "Backpack type" of type text
3. Go to Application Menu > API Builder
4. Add New API Application called "Test API"
5. Go to Schemas > Add New Schema > Name: "StudentSchema" Object: "Student" > Create
6. Go to Endpoints > Add API Endpoint > Retrieve Type: "Single Element" > Fill the other fields with mock data
7. Check that the endpoint is created
8. Response Body Schema > "StudentSchema"
     8.1. Check that path parameter property only shows External Reference Code, ID, and Carnet (the "Accept Unique Values Only" field)  > Select one of the three. If here the propertu Carnet is not shown, it may be necesary to reindex elastisearch (Application Menu > Control Panel > Configuration > Search > Index Actions > All Search Indexes > Reindex) 
9. Check that the endpoint can be updated 

**Screenshots:**
![image](https://github.com/liferay-headless/liferay-portal/assets/40394495/019e5d44-307f-4c67-89d8-def581a41847)
![image](https://github.com/liferay-headless/liferay-portal/assets/40394495/312f8e55-07f1-4593-992f-9f3ad12c4cf7)
![image](https://github.com/liferay-headless/liferay-portal/assets/40394495/231991bc-2f99-41d4-8576-fd9f2392ab85)

**Jira Ticket:** https://liferay.atlassian.net/browse/LPS-199304
